src/Spryker/Zed/AppWebhook/Business/WebhookHandler/WebhookHandler.php

```diff
@@ -9,30 +9,62 @@
 
 use Generated\Shared\Transfer\WebhookRequestTransfer;
 use Generated\Shared\Transfer\WebhookResponseTransfer;
+use Spryker\Zed\AppWebhook\Business\Identifier\IdentifierBuilderInterface;
+use Spryker\Zed\AppWebhook\Persistence\AppWebhookEntityManagerInterface;
+use Throwable;
 
 class WebhookHandler
 {
     /**
      * @param array<\Spryker\Zed\AppWebhook\Dependency\Plugin\WebhookHandlerPluginInterface> $webhookHandlerPlugins
      */
-    public function __construct(protected array $webhookHandlerPlugins)
-    {
+    public function __construct(
+        protected array $webhookHandlerPlugins,
+        protected AppWebhookEntityManagerInterface $appWebhookEntityManager,
+        protected IdentifierBuilderInterface $identifierBuilder
+    ) {
     }
 
     public function handleWebhook(WebhookRequestTransfer $webhookRequestTransfer, WebhookResponseTransfer $webhookResponseTransfer): WebhookResponseTransfer
     {
-        foreach ($this->webhookHandlerPlugins as $webhookHandlerPlugin) {
-            if (!$webhookHandlerPlugin->canHandle($webhookRequestTransfer)) {
-                continue;
-            }
+        // get the default identifier which will be only used when the webhook is not a retry
+        $identifier = $this->identifierBuilder->getIdentifier();
 
-            $webhookResponseTransfer = $webhookHandlerPlugin->handleWebhook($webhookRequestTransfer, $webhookResponseTransfer);
+        // Only persist the webhook if it's not a retry.
+        if ($webhookRequestTransfer->getIsRetry() !== true) {
+            $webhookRequestTransfer->setIdentifier($identifier);
+            $this->appWebhookEntityManager->saveWebhookRequest($webhookRequestTransfer);
         }
```

Comment:
I think it's better to save only if the request can't be handled by any of webhook plugins. In this way we can avoid overloaded webhook_inbox table.

```
if (!$webhookHandlerPlugin->canHandle()) { saveWebhookRequest() )

Reply:
The `canHandle` method will not tell anything. It is used to ask a specific module implementation if it can handle the current webhook e.g. the `AccountUpdated` webhook can only be handled by the AppMerchant package but not by the AppPayment. To prevent issues the `canHandle` method is used.

We don't know anyhow if there is an issue with any of the webhook handlers. no matter if it's failing because of an issue or exception or if it failed because of not handleable now but later.

I see your point in not overloading the database but this way we have the most possible robustness and fault-tolerance possible.

In case, we see a problem with it we can think about possible ways to improve.

For now, I'd like to stick with this solution.